python/ray/air/predictors/integrations/huggingface/huggingface_predictor.py

```diff
@@ -150,14 +150,14 @@ def predict(
 
 
         Returns:
-            DataBatchType: Prediction result.
+            Prediction result.
         """
-        if self.preprocessor:
-            data = self.preprocessor.transform_batch(data)
-
-        if isinstance(data, np.ndarray):
-            # If numpy array, then convert to pandas dataframe.
-            data = pd.DataFrame(data)
+        if TENSOR_COLUMN_NAME in data:
+            arr = data[TENSOR_COLUMN_NAME].to_numpy()
```

Comment:
This is a bit weird, but basically TensorArray prevents us from doing any kind of column selection, so in those cases we have to convert back to numpy, select the columns, and then go back to pandas.